presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -26,9 +29,110 @@ public TestTpchDistributedStats()
     {}
 
     @Test
-    void testEstimateForSimpleQuery()
+    void testTableScanStats()
     {
-        assertEquals(compareMetric("SELECT * FROM NATION").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM nation").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM part").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM partsupp").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM region").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM supplier").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM lineitem").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM lineitem").result(), MATCH);
     }
 
+    @Test
+    void testFilter()
+    {
+        String query = "" +
+                "SELECT * " +
+                "FROM lineitem " +
+                "WHERE l_shipdate <= DATE '1998-12-01' - INTERVAL '90' DAY";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), DIFFER);
+        assertEquals(metricComparison.getEstimate().get(), 30000, 1);
+    }
+
+    @Test
+    void testJoin()
+    {
+        String query = "SELECT * FROM  part, partsupp WHERE p_partkey = ps_partkey";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), DIFFER);
+        assertEquals(metricComparison.getEstimate().get(), 16000, 1);
+    }
+
+    @Test
+    void testUnion()
+    {
+        String query = "SELECT * FROM nation UNION SELECT * FROM nation";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), NO_ESTIMATE);
+    }
+
+    @Test
+    void testIntersect()
+    {
+        String query = "SELECT * FROM nation INTERSECT SELECT * FROM nation";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), NO_ESTIMATE);
+    }
+
+    @Test
+    void testExcept()
+    {
+        String query = "SELECT * FROM nation EXCEPT SELECT * FROM nation";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), NO_ESTIMATE);
+    }
+
+    @Test
+    void testEnforceSingleRow()
+    {
+        String query = "SELECT (SELECT max(n_regionkey) FROM nation)";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), MATCH);
+    }
+
+    @Test
+    void testValues()
+    {
+        String query = "VALUES 1";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), MATCH);
```

Comment:
Please extract an `assertMetricMatch` method, so that each test can be just:

```java
assertMetricMatch("SQL QUERY", OUTPUT_ROW_COUNT, equalTo(42).withAbsTolerance(1));
```